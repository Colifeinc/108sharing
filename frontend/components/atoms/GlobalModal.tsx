import * as React from 'react'
import Button from './Button'
import ModalLayout from 'components/layouts/ModalLayout'

interface IWindow {
  globalModal: GlobalModal
}
declare let window: IWindow

interface IProps {}

interface IState {
  title: string
  showModal: boolean
  body: React.ReactElement
  closeText: string
  submitText: string
  handleSubmit: () => void
  showSubmit?: boolean
}

interface IOpenModalProps {
  title?: string
  body: React.ReactElement
  closeText?: string
  submitText?: string
  handleSubmit?: () => void
  showSubmit?: boolean
}

const initialState = {
  title: '',
  showModal: false,
  body: null,
  closeText: '',
  submitText: '',
  handleSubmit: () => {},
  showSubmit: true,
}

export default class GlobalModal extends React.PureComponent<IProps, IState> {
  constructor(props) {
    super(props)

    this.state = { ...initialState }
    window.globalModal = this
  }

  public showModal = (config: IOpenModalProps) => {
    this.setState({
      ...initialState,
      ...config,
      showModal: true,
    })
  }

  public closeModal = () => {
    this.setState({ showModal: false })
  }

  public render() {
    return (
      <ModalLayout
        isOpen={this.state.showModal}
        closeModal={this.closeModal}
        title={this.state.title}
      >
        <>
          {this.state.body}
          {(this.state.closeText || this.state.handleSubmit) && this.state.showSubmit && (
            <div className="flex justify-center mt-6">
              {this.state.closeText && (
                <Button size="small" buttonType="muted" handleClick={this.closeModal}>
                  {this.state.closeText}
                </Button>
              )}
              {this.state.handleSubmit && (
                <Button
                  className="ml-4"
                  size="small"
                  buttonType="primary"
                  handleClick={this.state.handleSubmit}
                >
                  {this.state.submitText}
                </Button>
              )}
            </div>
          )}
        </>
      </ModalLayout>
    )
  }
}
