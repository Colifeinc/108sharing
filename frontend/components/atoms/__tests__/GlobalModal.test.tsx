import * as React from 'react'
import GlobalModal from '../GlobalModal'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('<GlobalModal />', () => {
  it('matches to the snapshot', () => {
    const wrapper = shallow(<GlobalModal />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
