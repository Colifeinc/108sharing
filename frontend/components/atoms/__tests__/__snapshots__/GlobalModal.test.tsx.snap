// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalModal /> matches to the snapshot 1`] = `
<ModalLayout
  closeModal={[Function]}
  isOpen={false}
  title=""
>
  <div
    className="flex justify-center mt-6"
  >
    <Button
      buttonType="primary"
      className="ml-4"
      handleClick={[Function]}
      size="small"
    />
  </div>
</ModalLayout>
`;
